lib/seattleflu/id3c/cli/command/etl/redcap_det_childcare.py

```diff
@@ -0,0 +1,535 @@
+        return None
+
+    location_resource_entries = build_residential_location_resources(
+        db = db,
+        cache = cache,
+        housing_type = enrollment.get('core_housing_type'),
+        primary_street_address = enrollment['core_home_street'],
+        secondary_street_address = enrollment['core_apartment_number'],
+        city = enrollment['core_home_city'],
+        state = enrollment['core_home_state'],
+        zipcode = enrollment['core_zipcode'],
+        system_identifier = INTERNAL_SYSTEM)
+
+    persisted_resource_entries = [patient_entry, *location_resource_entries]
+
+    childcare_center = enrollment['childcare_center']
+
+    for redcap_record_instance in redcap_record_instances:
+
+        event_type = None
+        study_arm = None
+
+        if redcap_record_instance.event_name.startswith(ENROLLMENT_EVENT_NAME_PREFIX):
+            event_type = EventType.ENROLLMENT
+        elif redcap_record_instance.event_name.startswith(ENCOUNTER_EVENT_NAME_PREFIX):
+            event_type = EventType.ENCOUNTER
+        else:
+            LOG.error(f'The record instance has an unexpected event name: {redcap_record_instance.event_name}')
+            continue
+
+        if '_arm_1' in redcap_record_instance.event_name:
+            study_arm = StudyArm.PRIMARY
+        elif '_arm_2' in redcap_record_instance.event_name:
+            study_arm = StudyArm.SECONDARY
+        else:
+            LOG.error(f'The record instance has an unexpected study arm in the event name: {redcap_record_instance.event_name}')
+            continue
+
+        # Skip an ENCOUNTER instance if we don't have the data we need to
+        # create an encounter. Require the participant to have provided
+        # survey data or a sample.
+        if event_type == EventType.ENCOUNTER \
+            and not is_complete('symptom_check', redcap_record_instance) \
+            and not is_complete('swab_kit_reg', redcap_record_instance) \
+            and not is_complete('post_collection_data_entry_qc', redcap_record_instance):
+            LOG.debug('Skipping record instance with insufficient information to construct the encounter')
+            continue
+
+        # From this point on, log at the `warning` level if we have to skip the encounter.
+        # That situation would be one we'd need to dig into.
+
+        # Create the site reference for the encounter. For primary participants, use
+        # a completed `return_pickup` survey to indicate that they are having their
+        # sample picked up from home instead of returning it to a dropbox.
+        location = None
+
+        site_map = {
+            'childcare_room_70th': SANDPOINT_SITE,
+            'childcare_room_radford' : RADFORD_SITE
+        }
+
+        if study_arm == StudyArm.SECONDARY or \
+                (study_arm == StudyArm.PRIMARY and is_complete('return_pickup', redcap_record_instance)):
+            location = None
+        else:
+            location = childcare_center
```

Comment:
I made the change in the "FIXUP: jover's feedback" commit.

Reply:
Sorry I wasn't clear in my original comment, I meant this can just be a single if statement:
```python
location = None
if study_arm == StudyArm.PRIMARY and not is_complete('return_pickup', redcap_record_instance):
    location = childcare_center
```
No need to assign to `None` value again since the `location` variable is declared with a default `None` value.

Reply:
Gotcha. I cleaned that up in the fixup.